tools/djxl_ng_main.cc

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "gflags/gflags.h"
+#include "jxl/codestream_header.h"
+#include "jxl/decode.h"
+#include "jxl/decode_cxx.h"
+#include "jxl/resizable_parallel_runner_cxx.h"
+#include "jxl/thread_parallel_runner.h"
+#include "jxl/thread_parallel_runner_cxx.h"
+#include "lib/extras/dec/decode.h"
+#include "lib/jxl/base/printf_macros.h"
+
+DECLARE_bool(help);
+DECLARE_bool(helpshort);
+
+DEFINE_int64(num_reps, 1,  // TODO(firsching): Clarify meaning of this
+                           // docstring. Is this simply for benchmarking?
+             "How many times to decompress.");
+
+DEFINE_int64(num_threads, 0,
+             // TODO(firsching): Sync with team about changed meaning of 0 -
+             // was: No multithreaded workers. Is: use default number.
+             "Number of worker threads (0 == use machine default).");
+
+// TODO(firsching): wire this up.
+DEFINE_bool(print_profile, false, "Print timing information before exiting.");
+
+// TODO(firsching): wire this up.
+DEFINE_int32(bits_per_sample, 0, "0 = original (input) bit depth");
+
+// TODO(firsching): wire this up.
+DEFINE_bool(
+    tone_map, true,
+    "tone map the image to the luminance range indicated by --display_nits "
+    "instead of performing a naive 0-1 -> 0-1 conversion");
+
+// TODO(firsching): wire this up.
+DEFINE_string(display_nits, "0.f-255.",
+              "luminance range of the display to which to "
+              "tone-map; the lower bound can be omitted");
+
+// TODO(firsching): wire this up.
+DEFINE_double(preserve_saturation, 0.1,
+              "with --tone_map, how much to favor saturation over luminance");
+
+// TODO(firsching): wire this up.
+DEFINE_string(color_space, "RGB_D65_SRG_Rel_Lin",
```

Comment:
I think the default ought to be the empty string -- and that the docstring is slightly unclear

Reply:
Added remark to consider it, when wiring up the flag.